docs/install.rst

```diff
@@ -1,115 +1,146 @@
+Then select your platform to install miniforge:
 
-.. _anaconda_install:
+.. tab-set::
 
-Using Conda
-===========
+    .. tab-item:: Linux & Mac
+        :sync: platform
 
-To install ``astropy`` using conda run::
+        Linux & Mac Run the script downloaded above, with
+        ``bash <filename>``. The following should work:
 
-    conda install astropy
+        .. code-block:: console
 
-``astropy`` is installed by default with the `Anaconda Distribution
-<https://www.anaconda.com/download/>`_. To update to the latest version run::
+            bash Miniforge3-$(uname)-$(uname -m).sh
 
-    conda update astropy
+        Once the installer has completed, close and reopen your terminal.
 
-There may be a delay of a day or two between when a new version of ``astropy``
-is released and when a package is available for conda. You can check
-for the list of available versions with ``conda search astropy``.
+    .. tab-item:: Windows
+        :sync: platform
 
-If you want to install ``astropy`` along with recommended or all of the
-available optional :ref:`dependencies <astropy-main-req>`, you can do::
+        Double click the executable file downloaded from
+        the links above.
 
-    conda install --channel conda-forge --channel defaults scipy matplotlib
+        Once the installer has completed you should have a new "miniforge
+        Prompt" entry in your start menu.
 
-or::
+In a new terminal (miniforge Prompt on Windows) run ``conda list`` to test that the install has worked.
 
-    conda install --channel conda-forge --channel defaults scipy matplotlib \
-      h5py beautifulsoup4 html5lib bleach pandas sortedcontainers \
-      pytz setuptools mpmath bottleneck jplephem asdf-astropy pyarrow
+Create Python Environment
+=========================
 
-To also be able to run tests (see below) and support :ref:`builddocs` use the
-following. We use ``pip`` for these packages to ensure getting the latest
-releases which are compatible with the latest ``pytest`` and ``sphinx`` releases::
+To create a new Python environment for ``astropy`` and other packages, start by
+launching a terminal (under a UNIX-like system) or the miniforge Prompt (under Windows).
+Now we will create and activate a new virtual environment to install ``astropy`` into:
 
-    python -m pip install pytest-astropy sphinx-astropy
+.. code-block:: bash
 
-.. warning::
+    $ conda create --name astropy python
+    $ conda activate astropy
 
-    Attempting to use `pip <https://pip.pypa.io>`__ to upgrade your installation
-    of ``astropy`` itself may result in a corrupted installation.
+In this case the environment we have created is named ``astropy`` but you can use any
+name you like.
 
-.. _testing_installed_astropy:
+In the future when you make a new terminal, you will need to run ``conda activate
+astropy`` to activate this environment.
 
-Testing an Installed ``astropy``
-================================
+.. _installing-astropy-with-pip:
```

Comment:
It may have been discussed previously but I don't think I get the flow we're going for here: are we encouraging users to use conda *and* pip to install astropy ? This seems like a recipe for disaster. In fact, I happen to have a screenshot from an older version of astropy's documentation that explicitly warned against it
![IMAGE 2024-09-10 19:25:36](https://github.com/user-attachments/assets/727f2dbb-5f78-4bd6-82ae-e4c9c4d49830)

Reply:
The plan is to create a vanilla conda environment with only Python and about a dozen other core packages that come along for the ride. After that, use `pip` for everything. I think this is quite safe.

What *might* cause issues is installing astropy from conda and then upgrading with pip, or vice versa. But it's worth mentioning that I think conda has gotten better with this mixed-method installation, at least in my experience. I regularly pip-install packages over a conda version and haven't had any problems recently. That doesn't mean it can't happen, but it hasn't happened to me in the last ~5 years.

Reply:
This strategy is basically what's in the developer docs. Going all-in with `pip` once the environment is created should minimize issues.

Reply:
> The plan is to create a vanilla conda environment with only Python and about a dozen other core packages that come along for the ride. After that, use pip for everything. I think this is quite safe.

In my experience people tend to follow instructions for environment creation and *stick with those envs*, so I'm not sure it's safe to assume that the only evolution that their env will see is what our instructions guide them through.
But in general I don't think conda solves any of *my* problems, hence I'm not comfortable with recommending anything other than pip (and maybe `uv` these days but I wouldn't through that one at a beginner yet). In short: I'm not sure I can help with this conversation.

Reply:
> But in general I don't think conda solves any of my problems

Conda has solved these problems for me:
- Installing any version of Python on Mac (or any of our supported platforms) in a lightweight and easily reversible way.
- Easily installing `graphviz` on Mac to allow building the docs.
- Installing Python on Windows and providing a capable shell that allows using that Python from a terminal.

Reply:
I am generally -1 on getting people to use pip inside a conda env for non-developers. I think it's asking them to use more tools together in a way which might cause trouble if they also need packages which don't have wheels in their environment in addition to astropy. It just feels over the top and complex, just because a few of us are happily doing that doesn't mean it's the best option for "regular users".

Reply:
I think that most conda / Anaconda users (millions?) have been mixing pip and conda for many years. Conda users go out to install a new package and encounter the instructions to `pip install` it and that's what happens.

What is the alternative solution that works uniformly for all platforms? The latest MacOS /usr/bin/python3 is version 3.9, which is not acceptable. ??

Reply:
I think our install instructions for release astropy should get people to install astropy from conda-forge. If something isn't on conda-forge then sure, use pip, but that's more friction and more chances for something to go wrong than sticking to a single package manager where possible in my opinion.

Reply:
For example, what happens if someone makes a conda env, then installs astropy from pip and then tries to conda install sunpy (which is what our install instructions say)?

Reply:
> The latest MacOS /usr/bin/python3 is version 3.9, which is not acceptable. ??

I don't think anyone is suggesting to recommend using *system* pythons. Personnaly my money is on `uv` as the next universal solution (it replaced a lot of the stack I previously used, like `pyenv`, `virtualenv` and `pipx`), but while it feels too early to recommend it in docs, it's a one-tool-fits-all.

Realistically, it might be that `conda` is still going to be how most scientific Python users set up their env in the forseeable future, and while the purist in me would prefer we avoid recommending mixing tools, it's probably an acceptable solution for now, but it should be clearly advertised that it has caveats.

Reply:
Dang, you have a point. I tested and `conda` wants to install everything even though most the dependencies are already there.

I still am convinced that making the base environment `conda create -n myenv python` and then managing with pip is 100% safe.

However, I'm not quite sure how to solve the problem of accidentally installing a boatload of core `conda` packages later. We certainly can expand the instructions to include a `conda` branch and a `pip` branch and highlight using one or the other.

The reason I really wanted to git rid of the `conda` install branch is the dependencies. It is so easy and concise to choose your optional dependency stack with `pip`. OTOH, with `conda` it is a pain (e.g. the equivalent of `pip install astropy[all]`) and requires ensuring that the install docs are maintained to agree with `pyproject.toml`.

Reply:
I don't understand what the problem is with people just `conda install astropy`?

Reply:
> I don't understand what the problem is with people just using conda install astropy?

That doesn't install the many optional dependencies, which are used very liberally in astropy. Basically if you do `conda install astropy` and then start trying to use it for any real task, you will have an on-going unpleasant experience of discovering one by one the dependencies that are missing. 

I know from personal experience when I only used conda how annoying that can be. We can certainly address that by providing the explicit conda packages like we do now:
```
conda install --channel conda-forge scipy matplotlib \
  h5py beautifulsoup4 html5lib bleach pandas sortedcontainers \
  pytz setuptools mpmath bottleneck jplephem asdf-astropy pyarrow
```
This is just ugly and prone to breakage. Is there a better way?

Reply:
> This is just ugly and prone to breakage.

I would say "explicit and robust" 😄 
(but it does add some maintenance burden)

Reply:
Well I think that's a bug with the conda recipe, see https://github.com/conda-forge/astropy-feedstock/pull/137 which was the minimal suggestion. Given the lack of optional dependencies in conda the general approach is to list all optional dependencies, which I am not sure why this isn't the case.

Reply:
> Is there a better way?

Use linux :smiley_cat:? (sorry, couldn't resist, but, really, it is sad how the lessons they learned from package management have been ignored by the python ecosystem)

More to the point, most users I've seen behave like those @neutrinoceros described, create an environment and stick with it, so the example of @cadair of a later `conda install` is indeed worrying. But that perhaps also means that giving the very long `conda install` comman is not such a big deal.

I guess my suggestion would be for the `conda` baseline to be just that, and then use that long line as a reason to describe the pip install as an alternative, one that is perhaps especially good if one's system python is actually recent.